standalone-integration/src/test/java/com/ericsson/bss/cassandra/ecchronos/standalone/ITOnDemandRepairJob.java

```diff
@@ -0,0 +1,318 @@
+
+    private OnDemandRepairSchedulerImpl myRepairSchedulerImpl;
+
+    private ScheduleManagerImpl myScheduleManagerImpl;
+
+    private CASLockFactory myLockFactory;
+
+    private List<String> myCreatedKeyspaces = new ArrayList<>();
+
+    @Before
+    public void init()
+    {
+        myLocalHost = getNativeConnectionProvider().getLocalHost();
+        mySession = getNativeConnectionProvider().getSession();
+        Cluster cluster = mySession.getCluster();
+        myMetadata = cluster.getMetadata();
+
+        myHostStates = HostStatesImpl.builder()
+                .withRefreshIntervalInMs(1000)
+                .withJmxProxyFactory(getJmxProxyFactory())
+                .build();
+
+        myRepairHistoryProvider = new RepairHistoryProviderImpl(mySession, s -> s, TimeUnit.DAYS.toMillis(30));
+
+        myLockFactory = CASLockFactory.builder()
+                .withNativeConnectionProvider(getNativeConnectionProvider())
+                .withHostStates(myHostStates)
+                .withStatementDecorator(s -> s)
+                .build();
+
+        myScheduleManagerImpl = ScheduleManagerImpl.builder()
+                .withLockFactory(myLockFactory)
+                .withRunInterval(100, TimeUnit.MILLISECONDS)
+                .build();
+        ReplicationState replicationState = new ReplicationState(myMetadata, myLocalHost);
+        myRepairSchedulerImpl = OnDemandRepairSchedulerImpl.builder()
+                .withJmxProxyFactory(getJmxProxyFactory())
+                .withTableRepairMetrics(myTableRepairMetrics)
+                .withScheduleManager(myScheduleManagerImpl)
+                .withRepairLockType(RepairLockType.VNODE)
+                .withReplicationState(replicationState)
+                .build();
+    }
+
+    @After
+    public void clean()
+    {
+        for (String keyspace : myCreatedKeyspaces)
+        {
+            for (TableMetadata tableMetadata : myMetadata.getKeyspace(keyspace).getTables())
+            {
+                mySession.execute(QueryBuilder.delete()
+                        .from("system_distributed", "repair_history")
+                        .where(QueryBuilder.eq("keyspace_name", keyspace))
+                        .and(QueryBuilder.eq("columnfamily_name", tableMetadata.getName())));
+            }
+
+            mySession.execute(SchemaBuilder.dropKeyspace(keyspace).ifExists());
+        }
+        myHostStates.close();
+        myRepairSchedulerImpl.close();
+        myScheduleManagerImpl.close();
+        myLockFactory.close();
+    }
+
+    /**
+     * Create a table that is replicated and schedule a repair on it
+     */
+    @Test
+    public void repairSingleTable()
+    {
+
+        TableReference tableReference = new TableReference("ks", "tb");
+
+        createKeyspace(tableReference.getKeyspace(), 3);
+        createTable(tableReference);
+
+        myRepairSchedulerImpl.scheduleJob(tableReference);
+
+        long startTime = System.currentTimeMillis();
```

Comment:
I think this should be at "start of method" or at least before #scheduleJob() to avoid potential race conditions.

Reply:
Done